example_code/shmem_team_context.c

```diff
@@ -0,0 +1,105 @@
+#include <shmem.h>
+#include <stdio.h>
+
+int isum, ival;
+
+int my_ctx_translate_pe(shmem_ctx_t src_ctx, int src_pe, shmem_ctx_t dest_ctx)
+{
+  shmem_team_t src_team, dest_team;
+  shmem_ctx_get_team(src_ctx, &src_team);
+  shmem_ctx_get_team(dest_ctx, &dest_team);
+  return shmem_team_translate(src_team, src_pe, dest_pe);
+}
+
+shmem_ctx_t my_team_create_ctx(shmem_team_t team) {
+  if (team == SHMEM_TEAM_NULL) {
+    return SHMEM_CTX_INVALID;
+  }
+
+  shmem_ctx_t ctx;
+  if (shmem_team_create_ctx(team, 0, &ctx) != 0) {
+    printf ("Failed to create context PE team!\n");
+    shmem_global_exit(-1);
```

Comment:
Could we return `SHMEM_CTX_INVALID` here instead of terminating the job?

Reply:
This seams good since we are encouraging team behavior that doesn't terminate the job. I changed this example so that it will complete in any case and just print messages to stderr if context creation fails at any point.